packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -285,6 +274,150 @@ export function ChatSettings(props: ChatSettingsProps): JSX.Element {
     );
   }
 
+  let languageModelSection;
+  if (lmProvider?.chat_models && lmProvider.chat_models.length > 0) {
```

Comment:
This needs to be iterated across all `server.lmProviders`, not only the current one, in a similar why to how the menu options are populated, see:

```ts
{server.lmProviders.providers.map(lmp =>
  lmp.models
    .filter(lm => lmp.chat_models.includes(lm))
    .map(lm => (
      <MenuItem value={`${lmp.id}:${lm}`}>
        {lmp.name} :: {lm}
      </MenuItem>
    ))
)}
```

Reply:
This should be fixed now.